app/src/main/java/org/projectbuendia/client/ui/BaseActivity.java

```diff
@@ -205,137 +207,195 @@ public void snackBarDismiss(@StringRes int[] id) {
+                            @Override public void onClick(View view) {
+                                // TODO: Display the actual server URL that couldn't be reached in
+                                // this message. This will require that injection be hooked up
+                                // through to
+                                // this inner class, which may be complicated.
+                                showMoreInfoDialog(
+                                    getString(R.string.troubleshoot_server_not_responding),
+                                    getString(R.string.troubleshoot_server_not_responding_details),
+                                    false);
+                            }
+                        }, 999, false);
+                    break;
+                case CHECK_PACKAGE_SERVER_REACHABILITY:
+                    snackBar(R.string.troubleshoot_package_server_unreachable,
+                        R.string.troubleshoot_action_more_info,
+                        new View.OnClickListener() {
+                            @Override public void onClick(View view) {
+                                showMoreInfoDialog(
+                                    getString(R.string.troubleshoot_package_server_unreachable),
+                                    getString(R.string.troubleshoot_update_server_unreachable_details),
+                                    true);
+                            }
+                        }, 998, false);
+                    break;
+                case CHECK_PACKAGE_SERVER_CONFIGURATION:
+                    snackBar(R.string.troubleshoot_package_server_misconfigured,
+                        R.string.troubleshoot_action_more_info,
+                        new View.OnClickListener() {
+                            @Override public void onClick(View view) {
+                                showMoreInfoDialog(
+                                    getString(R.string.troubleshoot_package_server_misconfigured),
+                                    getString(
+                                        R.string.troubleshoot_update_server_misconfigured_details),
+                                    true);
+                            }
+                        }, 999, false);
+                    break;
+                default:
+                    LOG.w("Troubleshooting action '%1$s' is unknown.", troubleshootingAction);
+                    return;
+            }
+        }
+    }
+
+    private void displayProblemSolvedMessage(HealthIssue solvedIssue) {
+        Map<HealthIssue, Integer[]> troubleshootingMessages = new HashMap<>();
+
+        // The troubleShootingMessages Map have the issue as the key and an integer array as
+        // value. This array is composed by:
+        // 0 - The message string id triggered by the issue.
+        // 1 - The new message string id (solved message).
+        // 2 - The timeout count for the solved message.
```

Comment:
For duration variables, it's important to put the units in the description, e.g. millis or seconds.